### ISSUE TITLE
Empty result_dir, cannot run KITTI evaluation

### ISSUE THREAD

**Issue by villanuevab** [kujason/avod#93]
In the README, there are instructions to use the KITTI evaluator that include: 
```
Copy the results folder into this folder. Each step should contain a 'data' folder.
```

However, I have no such results dir created after running run_training.py and run_evaluation.py. What and where is this results directory? 

I do have a `avod/scripts/offline_eval/results/` dir with a single file `avod_cars_example_results_0.1.txt` with contents: 
```
0
ERROR: Couldn't read: 002761.txt of ground truth. Please write me an email!
An error occured while processing your results.
1000
ERROR: Couldn't read: 002761.txt of ground truth. Please write me an email!
An error occured while processing your results.
2000
ERROR: Couldn't read: 002761.txt of ground truth. Please write me an email!
An error occured while processing your results.
3000
ERROR: Couldn't read: 002761.txt of ground truth. Please write me an email!
An error occured while processing your results.
4000
ERROR: Couldn't read: 002761.txt of ground truth. Please write me an email!
An error occured while processing your results.
5000
...
```
I inspected 002761.txt and it is identical to a fresh download of the file from the KITTI website, so I'm not sure what's going on... Your help is really appreciated! 

Thank you for your time.

**Comment by walzimmer** [comment]
"Copy the results folder into this folder. Each step should contain a 'data' folder."
The results folder is the predictions folder that is generated after running run_inference.py.
/avod/avod/data/outputs/avod_cars_example/predictions/final_predictions_and_scores/train/120000
'step' is the checkpoint that you want to evaluate (in my case the last checkpoint 120000).

Then you can run the file save_kitti_predictions.py to create labels in the following directories:
For 2d predictions:
avod/avod/data/outputs/avod_cars_example/predictions/kitti_predictions_2d/val/0.1/120000/data
For 3d predictions:
avod/avod/data/outputs/avod_cars_example/predictions/kitti_predictions_3d/val/0.1/120000/data

To evaluate the checkpoint run the following (adjust the paths to the label and the result folder):
./evaluate_object_3d_offline /KITTI/training/label_2 /avod/avod/data/outputs/avod_cars_example/predictions/kitti_predictions_2d/val/0.1/120000

I am getting the following values:
car_detection AP: 36.363636 35.893417 34.720699
car_orientation AP: 34.298126 32.345428 31.174282

Does someone know why those values are so low?
In the paper the following values were achieved:
car_detection AP:  73.59 65.78 58.38

I have used the default AVOD architecture and not the Feature Pyramid Network (AVOD-FPN).

**Comment by villanuevab** [comment]
Running 
```
./evaluate_object_3d_offline .../label_2/ ...avod/data/outputs/pyramid_cars_with_aug_example/predictions/final_predictions_and_scores/val/12000/
```
Only outputs 
```
done.
```

Is your output printed to a file? What is the expected output? 

Thanks for your time!

**Comment by walzimmer** [comment]
Your second path is still wrong. Pass the path to your kitti_predictions_2d or kitti_predictions_3d labels as I wrote in my first post:

Run the file save_kitti_predictions.py to create labels in the following directories:
For 2d predictions:
avod/avod/data/outputs/avod_cars_example/predictions/kitti_predictions_2d/val/0.1/119000/data
For 3d predictions:
avod/avod/data/outputs/avod_cars_example/predictions/kitti_predictions_3d/val/0.1/119000/data

To evaluate the checkpoint run the following (adjust the paths to the label and the result folder):
./evaluate_object_3d_offline /KITTI/training/label_2 /avod/avod/data/outputs/avod_cars_example/predictions/kitti_predictions_2d/val/0.1/119000

**Comment by villanuevab** [comment]
It worked this time. Thanks very much for your time. Seems I misunderstood the instructions. Here are my results for checkpoint 120000: 
```
pyramid_cars_with_aug_example/predictions/kitti_predictions_3d/val/0.1/120000/

car_detection AP: 90.070267 87.702118 80.443825
car_orientation AP: 89.976242 87.053314 79.664192
car_detection_BEV AP: 89.557861 86.561676 79.440514
car_heading_BEV AP: 89.456612 85.892670 78.634384
car_detection_3D AP: 77.769157 68.075890 67.329857
car_heading_3D AP: 77.726074 67.779854 66.830719
```

**Comment by yxy1995123** [comment]
I think I found the solution.In file run_eval.sh and run_eval_05_iou.sh../evaluate_object_3d_offline_05_iou ~/Kitti/object/training/label_2/ $2/$3 | tee -a ./$4_results_05_iou_$2.txt.
change the label path according to your computer

**Comment by patelrahil224** [comment]
In my case restarting the terminal worked! 
Not sure what the problem is. 

**When you see a source code working in the background there shouldn't be any problem as long as even the blank file lies in that directory.**


```
for (int32_t i=0; i<indices.size(); i++) {
    // file name
    char file_name[256];
    sprintf(file_name,"%06d.txt",indices.at(i));

    // read ground truth and result poses
    bool gt_success,det_success;
    vector<tGroundtruth> gt   = loadGroundtruth(gt_dir + "/" + file_name,gt_success);
    vector<tDetection>   det  = loadDetections(result_dir + "/data/" + file_name,
            compute_aos, eval_image, eval_ground, eval_3d, det_success);
    groundtruth.push_back(gt);
    detections.push_back(det);

    // check for errors
    if (!gt_success) {
      mail->msg("ERROR: Couldn't read: %s of ground truth. Please write me an email!", file_name);
      return false;
    }
    if (!det_success) {
      mail->msg("ERROR: Couldn't read: %s", file_name);
      return false;
    }
```

**Comment by Pari-singh** [comment]
Hi @villanuevab @walzimmer and authors,

I ran the evaluation on one of the checkpoints and for some reason in the end, I could not get the kitti evaluation and nothing was printed in results/__.txt file except for iteration numbers. The error I was getting was 
line 6: ./evaluate_object_3d_offline: No such file or directory

I tried giving the absolute path and yet it wasn't able to find the cpp file. Wanted to know from @villanuevab if you compiled your cpp file before running it as bash command?
